### PR TITLE
Create publishing workflow

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  publish:
+    permissions:
+      id-token: write # This is required for requesting the JWT
+    runs-on: ubuntu-latest
+    environment: pub-dev
+    steps:
+      - uses: actions/checkout@v3
+      - uses: isoos/setup-dart@oidc-token
```
Comment (sigurdm) [inline .github/workflows/publish.yml:16]: Not sure I like this specific line :)

Comment (jonasfj) [reply .github/workflows/publish.yml:16]: hehe, maybe it's time to merge it into upstream...

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+**'
```
Comment (jonasfj) [inline .github/workflows/publish.yml:6]: Do we need two stars? is this actually a regex pattern?

It's close I guess:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

Maybe, we just need `*` at the end...

**Review by jonasfj — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (sigurdm) [reply .github/workflows/publish.yml:6]: Yeah - one star is probably better

Comment (sigurdm) [inline .github/workflows/publish.yml:6]:
```suggestion
      - '[0-9]+.[0-9]+.[0-9]+*'
```

**Pull request merged**